specification/dartLangSpec.tex

```diff
@@ -16334,6 +16344,17 @@ \subsection{Identifier Reference}
   then $e$ evaluates to the function object obtained by closurization
   (\ref{functionClosurization})
   of $D$.
+\item
+  If $D$ is an instance getter declaration in an extension $E$,
+  then $e$ evaluates to the result of invoking said getter
+  with the current binding of \THIS{}, and
+  the current bindings of the type parameters declared by $D$.
+\item
+  If $D$ is an instance method declaration in an extension $E$
+  with type parameters \List{X}{1}{s},
+  then $e$ evaluates to the result of the extension method closurization
+  \code{$E$<\List{X}{1}{s}>(\THIS).\id}
+  (\ref{extensionMethodClosurization}).
```

Comment:
I'd say:

> ... to the tear-off of *D* closing over the current values of `this` and type parameters of *E*.

Do we need to handle instantiated tear-off here, or is that (implicitly) handled in type inference?

Reply:
About 'closing over'. We have explicitly specified the syntax `E<T1, .. Tk>(e).id` as an extension method closurization, don't you think we should stick to that syntax rather than switching to the less-explicitly defined 'closing over'?

As mentioned at line 16256, we have a general rule about generic function instantiation that we rely on in many locations.